thrust/detail/config/cpp_dialect.h

```diff
@@ -97,27 +97,43 @@
 #  define THRUST_COMP_DEPR_IMPL1 /* intentionally blank */
 #endif
 
-#define THRUST_COMPILER_DEPRECATION(REQ, FIX) \
-  THRUST_COMP_DEPR_IMPL(Thrust requires at least REQ. Please FIX. Define THRUST_IGNORE_DEPRECATED_CPP_DIALECT to suppress this message.)
+#define THRUST_COMPILER_DEPRECATION(REQ) \
+  THRUST_COMP_DEPR_IMPL(Thrust requires at least REQ. Define THRUST_IGNORE_DEPRECATED_CPP_DIALECT to suppress this message.)
+
+#define THRUST_COMPILER_DEPRECATION_SOFT(REQ, CUR) \
+  THRUST_COMP_DEPR_IMPL(Thrust requires at least REQ. CUR is deprecated but still supported. CUR support will be removed in a future release. Define THRUST_IGNORE_DEPRECATED_CPP_DIALECT to suppress this message.)
 
-// Minimum required compiler checks:
 #ifndef THRUST_IGNORE_DEPRECATED_COMPILER
+
+// Compiler checks:
 #  if THRUST_HOST_COMPILER == THRUST_HOST_COMPILER_GCC && THRUST_GCC_VERSION < 50000
-     THRUST_COMPILER_DEPRECATION(GCC 5.0, upgrade your compiler);
```

Comment:
Why do we no longer say "upgrade your compiler"?

Reply:
The message was getting too long and awkward with the requested additions, and `upgrade your compiler` is redundant with `your compiler is deprecated` so I dropped it.